File: backend/LibraryManagement.API/Models/Book.cs
using LibraryManagement.API.Enums;

namespace LibraryManagement.API.Models
{
    public class Book
    {
        public int Id { get; set; }

        public string Title { get; set; } = "";

        public string Author { get; set; } = "";

        public BookStatus Status { get; set; }
    }
}

File: src/types/Book.ts
export type BookStatus = "Available" | "Borrowed";

export interface Book {
  id: number;
  title: string;
  author: string;
  status: BookStatus;
}

export interface CreateBookInput {
  title: string;
  author: string;
}

export interface UpdateBookInput {
  title: string;
  author: string;
  status: BookStatus;
}